src/buildstructor/snapshots/buildstructor__buildstructor__codegen__tests__string_without_implicit_into.snap

```diff
@@ -0,0 +1,74 @@
+---
+source: src/buildstructor/codegen.rs
+expression: output
+---
+impl Foo {
+    pub fn builder() -> NewFooBuilder {
+        __foo_new_builder::new()
+    }
+}
+#[doc = "Autogenerated by buildstructor"]
+#[allow(type_alias_bounds)]
+pub type NewFooBuilder = __foo_new_builder::__FooBuilder<(__foo_new_builder::__Required<String>,)>;
+mod __foo_new_builder {
+    use super::*;
+    #[inline(always)]
+    pub fn new() -> __FooBuilder<(__foo_new_builder::__Required<String>,)> {
+        __FooBuilder {
+            fields: (__required(),),
+            _phantom: core::default::Default::default(),
+        }
+    }
+    pub struct __Required<T> {
+        _uninit: std::mem::MaybeUninit<T>,
+    }
+    pub struct __Optional<T> {
+        lazy: Option<T>,
+    }
+    pub struct __Set<T> {
+        value: T,
+    }
+    #[inline(always)]
+    fn __set<T>(value: T) -> __Set<T> {
+        __Set { value }
+    }
+    #[inline(always)]
+    fn __required<T>() -> __Required<T> {
+        __Required::<T> {
+            _uninit: std::mem::MaybeUninit::uninit(),
+        }
+    }
+    #[inline(always)]
+    fn __optional<T>() -> __Optional<T> {
+        __Optional::<T> { lazy: None }
+    }
+    impl<T: Default> From<__Optional<T>> for __Set<T> {
+        #[inline(always)]
+        fn from(o: __Optional<T>) -> Self {
+            __Set {
+                value: o.lazy.unwrap_or_default(),
+            }
+        }
+    }
+    pub struct __FooBuilder<__P> {
+        fields: __P,
+        _phantom: core::marker::PhantomData<()>,
+    }
+    impl __FooBuilder<(__Required<String>,)> {
+        #[inline(always)]
+        pub fn foo(self, foo: String) -> __FooBuilder<(__Set<String>,)> {
+            let foo = foo;
+            __FooBuilder {
+                fields: (__set(foo),),
+                _phantom: core::default::Default::default(),
+            }
+        }
+    }
+    impl<__P0: Into<__Set<String>>> __FooBuilder<(__P0,)> {
```

Comment:
note that currently, this PR still does not fix my issue, because of this remaining `Into<_>` wrapper... I'm currently trying to remove it optionally